common/cmake/ALPSCoreConfig.cmake.in

```diff
@@ -80,6 +80,27 @@ function(alpscore_config_main_)
         endif()
       endforeach()
     endif()
+
+    set(alps_boost_dir_ @Boost_DIR@)
+    set(alps_boost_root_ @BOOST_ROOT@)
+    set(alps_boost_includedir_ @Boost_INCLUDE_DIR@)
+    set(alps_boost_librarydir_ @Boost_LIBRARY_DIR_RELEASE@ @Boost_LIBRARY_DIR_DEBUG@)
+
+    if (alps_boost_dir_)
+      set(Boost_DIR ${alps_boost_dir_} PARENT_SCOPE)
+    endif()
+    if (alps_boost_root_)
+      set(BOOST_ROOT ${alps_boost_root_} PARENT_SCOPE)
+    endif()
+    if (alps_boost_includedir_)
+      set(BOOST_INCLUDEDIR ${alps_boost_includedir_} PARENT_SCOPE)
+    endif()
+    if (alps_boost_librarydir_)
+      list(REMOVE_DUPLICATES alps_boost_librarydir_)
+      set(BOOST_LIBRARYDIR ${alps_boost_librarydir_} PARENT_SCOPE)
```

Comment:
careful here and with the includedor: does the capitalization matter? Propose to change it so we have their silly spelling (BOOST_ROOT but Boost_INCLUDE_DIR)

Reply:
I guess, some clarification would be useful here. :)

In lines 84-87 we *remember* the current values of the variables that were set by the CMake Boost module  (so to say, "output varaibles"): `Boost_DIR`, `Boost_INCLUDE_DIR` and `Boost_LIBRARY_DIR_*`, with this capitalization. This happens at *ALPSCore* configure time. 

In lines 89-100 we *set* variables that affect the behavior of Boost module ("input variables"), hinting it to the location of Boost: `BOOST_ROOT`, `BOOST_INCLUDEDIR`, `BOOST_LIBRARYDIR`, with this capitalization. This happens at *user code* configure time.

The names of both input and output variables are defined by the [Boost module documentation](https://cmake.org/cmake/help/v3.1/module/FindBoost.html).